### DISCUSSION TITLE
What exactly does this plugin do?

### DISCUSSION THREAD

**Discussion by bendavis78** [WhoIsSethDaniel/mason-tool-installer.nvim#48] [Q&A] [answered]
I'm a bit confused as to what this does that isn't already provided by mason. What is meant by a "tool"? Is it just mason packages?

I'm personally looking for a way to manage python packages for various plugins, but that seems to be outside of the scope of Mason (for packages not explicitly in the Mason registry). Is that something this plugin can help with?

**Comment by serranomorante** [comment, marked as answer]
No, as far as I'm concerned, this plugin provides a convenient way of auto-installing/updating packages by using an `ensure_installed` table (which Mason doesn't provide) from the mason registry only. Additionally you can use mason naming convention `typescript-language-server` or lspconfig names like `tsserver`, or null-ls names or nvim-dap names.

**Reply by Gaweringo** [reply]
Just as a small note: [mason-lspconfig](https://github.com/mason-org/mason-lspconfig) does provide a `ensure_installed` option, so if you're just looking for that you only need mason-lspconfig.
mason-tool-installer seems to add more features around that, like setting a specific version,  autoupdate and alike.
